lib/active_model/serializer.rb

```diff
@@ -66,10 +66,14 @@ def serializer_for(resource, options = {})
             ArraySerializer
           end
         else
-          klass_name = build_serializer_class(resource, options)
-          Serializer.serializers_cache.fetch_or_store(klass_name) do
-            _const_get(klass_name)
-          end
+          search_list = build_serializer_class_list(resource, options)
```

Comment:
Upon review,, I should have noticed these changes are out of scope for this pr and asked for them to be proposed separately.